### PR TITLE
Update architecture documentation

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Update architecture documentation based on the articles published on build
| Fixed ticket? | ~

### CODE REVIEW

File: src/content/1.7/development/architecture/file-structure/_index.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+Contains bootstrapping and configuration files. Unless asked, you should *never* edit them, as they are directly handled by PrestaShop's installer and back office.
+
+### /controllers
+
+Contains controllers – as in Model-View-Controller (or MVC), the software architecture used by PrestaShop. Each file controls a specific part of PrestaShop.
+
+Controllers are separated in two directories:
+
+- `/admin`: contains legacy controllers for the Back Office
```
Comment (matks) [inline src/content/1.7/development/architecture/file-structure/_index.md:102]: Maybe we should provide a link to a page where we define what is the meaning of "legacy" ? Newcomers at PrestaShop may not understand it

File: src/content/1.7/development/architecture/file-structure/_index.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+Contains all HTML and text templates for e-mails sent by PrestaShop. Each language has its specific folder, where you can manually edit their content if you wish. PrestaShop contains a tool to edit your e-mails, located in the back office, in the Localization &gt; Translation page.
+
+Since {{< minver v="1.7.6" >}} this directory also contains [Mail themes]({{< ref "/1.7/development/components/mail-templates/_index.md" >}}).
+
+### /modules
+
+Contains all of PrestaShop's modules, each in its own folder. 
+
+{{% notice tip %}}
+If you wish to definitely remove a module, first uninstall it from the back office.
```
Comment (matks) [inline src/content/1.7/development/architecture/file-structure/_index.md:172]:
```suggestion
If you wish to definitely remove a module, it is better to trigger the uninstall process from the back office before deleting the module folder
```

File: src/content/1.7/development/architecture/file-structure/understanding-src-folder.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+4. It SHOULD follow SOLID principles, in particular the Dependency Inversion principle.
+5. It SHOULD NOT depend directly on the Symfony framework.
+6. It SHOULD NOT depend directly on Adapter classes (it SHOULD depend on interfaces instead).
+7. New interfaces MUST be added in Core, unless they depend on legacy code.
+
+## Adapter
+
+Adapter classes belong to the `PrestaShop\PrestaShop\Adapter` namespace.
+
+Since Core code cannot depend directly on legacy classes, classes in the Adapter namespace provide Core classes with "bridge" to legacy classes by following the [Adapter pattern strategy](https://en.wikipedia.org/wiki/Adapter_pattern). Through this pattern, Adapter classes wrap legacy code in a way that implements the interface required by Core.
```
Comment (matks) [inline src/content/1.7/development/architecture/file-structure/understanding-src-folder.md:36]: Same here, we might explain what is the meaning of "legacy classes".

File: src/content/1.7/development/architecture/file-structure/understanding-src-folder.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+4. It SHOULD follow SOLID principles, in particular the Dependency Inversion principle.
+5. It SHOULD NOT be a simple copy/paste of legacy code in a different namespace.
+
+You should create a new file here whenever you have a hard bound with legacy files, and
+you can't refactor it easily. For instance, take a look at `PhpParameters` class in "Configuration": this class doesn't rely on
+Context or specific constants but on the configuration files of PrestaShop legacy framework. As these files are shared by both front and back, we couldn't manage to refactor it and remove it now.
+
+## PrestaShopBundle
+
+Adapter classes belong to the `PrestaShopBundle` namespace. The bundle contains the "glue" between Symfony framework and PrestaShop business services.
```
Comment (matks) [inline src/content/1.7/development/architecture/file-structure/understanding-src-folder.md:54]:
```suggestion
Bundle classes belong to the `PrestaShopBundle` namespace. The bundle contains the "glue" between Symfony framework and PrestaShop business services.
```

Comment (matks) [reply src/content/1.7/development/architecture/file-structure/understanding-src-folder.md:54]:
```suggestion
Bundle classes belong to the `PrestaShopBundle` namespace. This folder contains the code composing the application layer, meaning:
- the classes wrapping PrestaShop business logic into a Symfony application
- the classes relying heavily on Symfony implementation such as Controllers, Twig templates, Form Types
```

File: src/content/1.7/development/architecture/file-structure/_index.md (hunk 2 above)
Comment (eternoendless) [reply src/content/1.7/development/architecture/file-structure/_index.md:172]: If I'm not mistaken when you uninstall a module from the back office you're given the option to keep or to remove the module's files.

```suggestion
If you wish to definitely remove a module, it is better to uninstall it from the back office.
```

Wdyt?

File: src/content/1.7/development/architecture/file-structure/understanding-src-folder.md (hunk 4 above)
Comment (eternoendless) [reply src/content/1.7/development/architecture/file-structure/understanding-src-folder.md:54]:
```suggestion
Bundle classes belong to the `PrestaShopBundle` namespace. This folder contains the code composing the application layer, meaning:

- Classes wrapping PrestaShop business logic into a Symfony application
- Classes relying heavily on Symfony such as Controllers, Twig templates, Form Types, etc.
```

File: src/content/1.7/development/architecture/file-structure/_index.md (hunk 1 above)
Comment (eternoendless) [reply src/content/1.7/development/architecture/file-structure/_index.md:102]: Legacy is explained in the introduction: https://github.com/PrestaShop/docs/blob/a87373a4f92b4cb6ab9545baf75e4a86dd1ed13d/src/content/1.7/development/architecture/introduction.md#L74-L79

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @eternoendless

**Pull request merged**